python/setup.py

```diff
@@ -391,7 +391,7 @@ def get_packages(self):
         "jsonschema",
         "msgpack >= 1.0.0, < 2.0.0",
         "packaging",
-        "protobuf >= 3.15.3, != 3.19.5",
+        "protobuf >= 5.28.0,<6.0.0",
```

Comment:
Why do we need to update the python protobuf version?

Reply:
In other words, staying up to date should not require moving up the lower bound for the build (modulo some long-term changes to the default format edition).

Though I appreciate that the devil is in the details, especially staging compatible across several C++ major versions (e.g. v6.30 changed the return type of several functions from `char*` to `std::string_view`)

Reply:
> the decision to raise the minimum

I'm of course completely fine with that (note that recent protobuf landed some changes to become compatible with java-protobuf 3.x again, but you're probably more familiar with that than me by now).

However, I don't see any changes to raise the minimum in this PR anymore. `python/setup.py` still has
https://github.com/ray-project/ray/blob/fefcc147a99102199e99a465fd38532c93155825/python/setup.py#L386
and isn't being changed here.